### PR TITLE
MSI-2593-Allow-to-filter-Store-Pickup-orders-on-order-grid.

### PR DESCRIPTION
**Pull request by swnsma**
### Description (*)
Add column to the Grid.

### Fixed Issues (if relevant)
1. https://github.com/magento/inventory/issues/2593: Allow to filter Store Pickup orders on order grid

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/InventoryInStorePickup/etc/events.xml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0"?>
+<!--
+  ~ Copyright © Magento, Inc. All rights reserved.
+  ~ See COPYING.txt for license details.
+  -->
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:Event/etc/events.xsd">
+    <event name="checkout_submit_all_after">
```
Comment (novikor) [inline app/code/Magento/InventoryInStorePickup/etc/events.xml:7]: Looks sad, but I assume you had to do it

File: app/code/Magento/InventoryInStorePickup/Observer/UpdateOrderGrid.php [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+    /**
+     * Update the Order Grid in case Pickup Location was added to the order.
+     *
+     * @param Observer $observer
+     * @return void
+     */
+    public function execute(Observer $observer)
+    {
+        if (!$this->globalConfig->getValue('dev/grid/async_indexing')) {
```
Comment (novikor) [inline app/code/Magento/InventoryInStorePickup/Observer/UpdateOrderGrid.php:59]: Use `\Magento\InventoryInStorePickup\Model\Order\GetPickupLocationCode` instead.

**Review by novikor — CHANGES_REQUESTED** [review]

File: app/code/Magento/InventoryInStorePickup/Observer/UpdateOrderGrid.php (hunk 2 above)
Comment (swnsma) [reply app/code/Magento/InventoryInStorePickup/Observer/UpdateOrderGrid.php:59]: Done

**Review by novikor — APPROVED** [review]

**Comment by novikor** [comment]
@magento run Static Tests Build

**Comment by novikor** [comment]
@magento run Unit Tests Build

**Comment by novikor** [comment]
@magento run all tests

**Comment by smoskaluk** [comment]
Hi @swnsma 
In *Stores* -> *Sources* from Inventory the check-boxes from :gear:**Columns* could not be checked-unchecked by clicking on checkbox *name**(text) only if click directly on checkbox square. This happens after creating a New View, and switching back to Default View
- seems like there is an issue with html `<label for="...">` atribute  
- not reproducible if perform the same steps in *Catalog* -> *Products* section
- the checkboxes can be checked-unchecked by clicking on checkbox **name**(text) again, if Refresh (F5) the web page

**Comment by swnsma** [comment]
@ishakhsuvarov @novikor, please do something.

**Pull request merged**